Ghidra/Features/Decompiler/src/main/java/ghidra/app/decompiler/DecompileCallback.java

```diff
@@ -1316,33 +1316,14 @@ public StringData getStringData(String addrString, String dtName, String dtId) {
 		}
 		Data data = program.getListing().getDataContaining(addr);
 		Settings settings = SettingsImpl.NO_SETTINGS;
-		AbstractStringDataType dataType = null;
 		StringDataInstance stringInstance = null;
 		int length = 0;
-		if (data != null) {
-			if (data.getDataType() instanceof AbstractStringDataType) {
-				// There is already a string here.  Use its configuration to
-				// set up the StringDataInstance
-				settings = data;
-				dataType = (AbstractStringDataType) data.getDataType();
-				length = data.getLength();
-				if (length <= 0) {
-					return null;
-				}
-				long diff = addr.subtract(data.getAddress()) *
-					addr.getAddressSpace().getAddressableUnitSize();
-				if (diff < 0 || diff >= length) {
-					return null;
-				}
-				length -= diff;
-				MemoryBufferImpl buf = new MemoryBufferImpl(program.getMemory(), addr, 64);
-				stringInstance = dataType.getStringDataInstance(buf, settings, length);
-			}
-		}
-		if (stringInstance == null) {
+		stringInstance = StringDataInstance.getStringDataInstance(data);
+		if (stringInstance == StringDataInstance.NULL_INSTANCE) {
 			// There is no string and/or something else at the address.
 			// Setup StringDataInstance based on raw memory
 			DataType dt = dtmanage.findBaseType(dtName, dtId);
+			AbstractStringDataType dataType = null;
```

Comment:
This local var (dataType) needs a better name, ie. tmpStringDataType

Reply:
This actually is the original name... I just followed the previous version. I thought it was better to keep the commit minimum.

Reply:
Fixed.